.github/workflows/ci.yml

```diff
@@ -0,0 +1,112 @@
+  test_golang_oldest:
+    name: Go tests with previous Go version
+    runs-on: ubuntu-latest
+    # The go verson in this image should be N-1 wrt test_go.
+    container:
+      image: quay.io/prometheus/golang-builder:1.19-base
+    steps:
+      - uses: actions/checkout@v3
+      - run: make build
+      - run: go test --tags=stringlabels ./...
+
+  build_all:
+    name: Build Prometheus for all architectures
+    runs-on: ubuntu-latest
+    if: |
+      (github.event_name == 'push')
+      ||
+      (github.event_name == 'pull_request')
+    strategy:
+      matrix:
+        thread: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11 ]
+
+    # Whenever the Go version is updated here, .promu.yml
+    # should also be updated.
+    steps:
+      - uses: actions/checkout@v3
+      - uses: prometheus/promci@v0.0.2
+      - uses: ./.github/promci/actions/build
+        with:
+          parallelism: 12
+          thread: ${{ matrix.thread }}
+  golangci:
+    name: golangci-lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.20.x
+      - name: Install snmp_exporter/generator dependencies
+        run: sudo apt-get update && sudo apt-get -y install libsnmp-dev
+        if: github.repository == 'prometheus/snmp_exporter'
+      - name: Lint
+        uses: golangci/golangci-lint-action@v3.4.0
+        with:
+          version: v1.51.2
+  fuzzing:
+    uses: ./.github/workflows/fuzzing.yml
+    if: github.event_name == 'pull_request'
+  codeql:
+    uses: ./.github/workflows/codeql-analysis.yml
+
+  publish_main:
+    name: Publish main branch artifacts
+    runs-on: ubuntu-latest
+    needs: [test_go, test_windows, golangci, codeql, build_all]
+    if: github.event_name == 'push' && github.event.ref == 'refs/heads/main'
+    steps:
+      - uses: actions/checkout@v3
+      - uses: prometheus/promci@v0.0.2
+      - uses: ./.github/promci/actions/publish_main
+        with:
+          docker_hub_login: ${{ secrets.docker_hub_login }}
+          docker_hub_password: ${{ secrets.docker_hub_password }}
+          quay_io_login: ${{ secrets.quay_io_login }}
+          quay_io_password: ${{ secrets.quay_io_password }}
+  publish_release:
+    name: Publish release artifacts
+    runs-on: ubuntu-latest
+    needs: [test_go, test_windows, golangci, codeql, build_all]
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v')
+    steps:
+      - uses: actions/checkout@v3
+      - uses: prometheus/promci@v0.0.2
+      - uses: ./.github/promci/actions/publish_release
+        with:
+          docker_hub_login: ${{ secrets.docker_hub_login }}
+          docker_hub_password: ${{ secrets.docker_hub_password }}
+          quay_io_login: ${{ secrets.quay_io_login }}
+          quay_io_password: ${{ secrets.quay_io_password }}
+          github_token: ${{ secrets.PROMBOT_GITHUB_TOKEN }}
```

Comment:
This should probably be `RICHIH_GITHUB_TOKEN`, since this is not going to publish to a Prometheus org.

Maybe we should drop the publishing steps for now and just have CI that runs.

Reply:
> This should probably be RICHIH_GITHUB_TOKEN, since this is not going to publish to a Prometheus org.

Will do!

> Maybe we should drop the publishing steps for now and just have CI that runs.

With the CircleCI version if `docker_hub_password` or `quay_io_login` was not present then it would simply skip those steps. I'm reasonably confident this will work.